packages/core/src/extension/extension-messenger.ts

```diff
@@ -0,0 +1,101 @@
+import { IframeEventPoster } from '../common/iframe';
+import { appendWww, parseUrl } from '../common/url';
+import { bindDecorator as bind } from '../common/utility';
+
+import { ExtensionNotFoundError } from './errors';
+import { Extension } from './extension';
+import { ExtensionCommandHandlers } from './extension-command-handler';
+import { ExtensionOriginEvent } from './extension-origin-event';
+import { HostOriginEvent } from './host-origin-event';
+
+export class ExtensionMessenger {
+    private _isListening = false;
+    private _extensions: Extension[] | undefined;
+    private _extensionCommandHandlers: ExtensionCommandHandlers = {};
+    private _posters: { [extensionId: string]: IframeEventPoster<HostOriginEvent> } = {};
+
+    listen(extensions: Extension[], extensionCommandHandlers: ExtensionCommandHandlers): void {
+        this._extensions = extensions;
+        this._extensionCommandHandlers = extensionCommandHandlers;
```

Comment:
It looks like a non-TS client can listen to / post any arbitrary command not defined in `ExtensionCommand`? It would be good to restrict that so that only known commands can be registered.

Reply:
Sure, I've implemented a filter to only handle enum ExtensionCommand In the extensionMessenger's `listen()` method.